middlewares/hacker.middleware.js

```diff
@@ -308,6 +309,29 @@ async function sendAppliedStatusEmail(req, res, next) {
     Services.Email.sendStatusUpdate(account.firstName, account.email, Constants.General.HACKER_STATUS_APPLIED, next);
 }
 
+/**
+ * Sends an email telling the user that they have applied. This is used exclusively when we POST a hacker.
+ * @param {{body: {hacker: {accountId: string}}}} req 
+ * @param {*} res 
+ * @param {(err?:*)=>void} next 
+ */
+async function sendTicketEmail(req, res, next) {
+    const hacker = req.body.hacker;
+    const address = Services.Env.isProduction() ? process.env.FRONTEND_ADDRESS_DEPLOY : process.env.FRONTEND_ADDRESS_DEV;
+    const httpOrHttps = (address.includes("localhost")) ? "http" : "https";
+    const singleHackerViewLink = Services.Hacker.generateHackerViewLink(httpOrHttps, address, hacker._id.toString());
+    const ticketSVG = await Services.Hacker.generateQRCode(singleHackerViewLink);
+    const account = await Services.Account.findById(hacker.accountId);
+    if (!account || !ticketSVG) {
+        return next({
+            status: 500,
+            message: Constants.Error.GENERIC_500_MESSAGE,
+            error: {}
+        });
+    }
+    Services.Email.sendTicketEmail(account.firstName, account.email, ticketSVG, next);
```

Comment:
Was there a reason to use callbacks here instead of await/async? It would be nice for our services to be consistent.

Reply:
The email library we use doesn’t support async / await, so back when I first created the service, I followed the library’s standards. I can try promisifying it and see what happens then.

Reply:
Ah I see - unfortunate

Reply:
Yeah :’(